examples/common_patterns_for_connectors/priority_first_sync_for_high_volume_initial_syncs/users_sync.py

```diff
@@ -0,0 +1,42 @@
+from datetime import timezone
+
+from fivetran_connector_sdk import Operations as op
+import mock_api
+import connector
+
+def sync_users(base_url, params, state, is_historical_sync):
+    more_data = True
+    last_updated_at = params['updated_since']
+
+    while more_data:
+        # Get response page from mock API call.
+        response_page = mock_api.get_api_response(base_url, params)
+
+        # Process the items.
+        items = response_page.get("data", [])
+        if not items:
+            break  # End pagination if there are no records in response.
+
+        for user in items:
+            yield op.upsert(table="user", data=user)
+            last_updated_at = user["updated_at"] # Assuming the API returns the data in ascending order
+
+        if is_historical_sync and connector.get_datetime_object(last_updated_at) >= connector.get_datetime_object(
+                connector.get_pfs_historical_cursor_for_endpoint(state, 'user')).astimezone(timezone.utc):
+            connector.set_pfs_historical_cursor_for_endpoint(state, 'user', params['updated_since'])
```

Comment:
Historical cursor should go backwards right? But if there are multiple pages then we are assigning `params['updated_since']` with `last_updated_at` which will then represent the latest timestamp, as you are assuming the data is in ascending order?

Reply:
Yep fixed it.